crates/sui-indexer-alt-framework/src/cluster.rs

```diff
@@ -50,69 +50,112 @@ pub struct IndexerCluster {
     cancel: CancellationToken,
 }
 
-impl IndexerCluster {
-    /// Create a new cluster with most of the configuration set to its default value. Use
-    /// [Self::new_with_configs] to construct a cluster with full customization.
-    pub async fn new(
-        database_url: Url,
-        args: Args,
-        migrations: Option<&'static EmbeddedMigrations>,
-    ) -> Result<Self> {
-        Self::new_with_configs(
-            database_url,
-            DbArgs::default(),
-            args,
-            IngestionConfig::default(),
-            migrations,
-            None,
-        )
-        .await
+/// Builder for creating an IndexerCluster with a fluent API
+#[derive(Default)]
+pub struct IndexerClusterBuilder {
+    database_url: Option<Url>,
+    db_args: DbArgs,
+    args: Option<Args>,
+    ingestion_config: IngestionConfig,
+    migrations: Option<&'static EmbeddedMigrations>,
+    metric_label: Option<String>,
+}
+
+impl IndexerClusterBuilder {
+    /// Create a new builder instance
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// Set the PostgreSQL database connection URL (required).
+    ///
+    /// This should be a valid PostgreSQL connection string, e.g.:
+    /// - `postgres://user:password@host:5432/mydb`
+    pub fn with_database_url(mut self, url: Url) -> Self {
+        self.database_url = Some(url);
+        self
     }
 
-    /// Create a new cluster.
+    /// Configure database connection parameters such as pool size, connection timeout, etc.
     ///
-    /// - `database_url` and `db_args` configure its database connection.
-    /// - `args` configures where checkpoints are come from, what is indexed and metrics.
-    /// - `ingestion_config` controls how the ingestion service is set-up (its concurrency, polling
-    ///    intervals, etc).
-    /// - `migrations` are any database migrations the indexer needs to run before starting to
-    ///   ensure the database schema is ready for the data that is about to be committed.
-    /// - `metric_label` is an optional custom label to add to metrics reported by this service.
-    pub async fn new_with_configs(
-        database_url: Url,
-        db_args: DbArgs,
-        args: Args,
-        ingestion_config: IngestionConfig,
-        migrations: Option<&'static EmbeddedMigrations>,
-        metric_label: Option<String>,
-    ) -> Result<Self> {
+    /// Defaults to `DbArgs::default()` if not specified, which provides reasonable defaults
+    /// for most use cases.
+    pub fn with_db_args(mut self, args: DbArgs) -> Self {
+        self.db_args = args;
+        self
+    }
```

Comment:
I actually thought you were going to expose a fluent API for this, `args`, and `IngestionConfig` also, because these are the structs that we are most likely to add new flags to in future.

What are your thoughts on adding those, probably in addition to a setter that allows you to set the full args struct, in case it comes from the command-line?